platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblems.kt

```diff
@@ -407,21 +405,17 @@ class ConfigurationCacheProblems(
         }
     }
 
-    private
-    fun areDegradationReasonsPresent(): Boolean = degradationController.hasDegradationReasons
-
-    private
-    fun areTaskDegradationReasonsPresent(): Boolean = degradationController.hasTaskDegradationReasons
-
     private
     fun degradationSummary(): String {
-        val degradingTaskCount = degradationController.degradedTaskCount
+        val degradingTaskCount = getDegradationDecision().degradedTaskCount
         val degradingFeatures = buildList {
-            degradationController.visitDegradedFeatures { feature, _ -> add(feature) }
+            getDegradationDecision().onDegradedFeature { feature, _ -> add(feature) }
         }
         return DegradationSummary(degradingFeatures, degradingTaskCount).render()
     }
 
+    private fun getDegradationDecision() = degradationController.degradationDecision
```

Comment:
💅 I'd make this a computed property `val degradationDecision get() = ...`. Use-sites will read better that way.